### DISCUSSION TITLE
Change scale of an object as you get closer

### DISCUSSION THREAD

**Discussion by outandaboutar** [AR-js-org/AR.js#267] [General]
Hi
I am trying to understand how you would resize the scale as you get closer.  Im still experimenting with one of the tutorials and I have 3 objects:

var models = [
    {
        url: './assets/magnemite/scene.gltf',
        scale: '0.5 0.5 0.5',
        info: 'Magnemite, Lv. 5, HP 10/10',
        rotation: '0 180 0',
        position: '50 0 -200',
    },
    {
        url: './assets/articuno/scene.gltf',
        scale: '0.2 0.2 0.2',
        rotation: '0 180 0',
        position: '50 0 -200',
        info: 'Articuno, Lv. 80, HP 100/100',
    },
    {
        url: './assets/dragonite/scene.gltf',
        scale: '0.08 0.08 0.08',
        rotation: '0 180 0',
        position: '50 0 -200',
        info: 'Dragonite, Lv. 99, HP 150/150',
    },
];

I am familiar with coding to some degree but i've never looked at AR before so I would just really appreciate some more help (thanks very much for previous help guys).  I imagine I would need to turn the scale attribute into a variable, and then create an if statement some where to check on users position and as the user moves closer to one of the objects, I can use the distance from the user to the object to effect the scale of the object.  But having multiple objects is messing with my mind and I just dont know how to access the users position?  Can this be accessed as a variable or something?

cheers!
Dan

**Comment by jonnypage** [comment]
Are you using geolocations or marker/NFTs?

**Reply by outandaboutar** [reply]
I believe I am using geolocations.  This is the tutorial i've used and I havent changed it really, other than to create a location outside my house to test.
https://medium.com/chialab-open-source/build-your-location-based-augmented-reality-web-app-c2442e716564

**Reply by nickw1** [reply]
Objects should automatically get larger as you move closer to them and smaller as you move away from them. Is that what you mean?

Is it not occurring in your case? It may be related to #278.
Try using this to test: https://hikar.org/265/.

This will create two cubes a short distance to the north of your initial GPS position. Try moving in a direction other than north (at least 100m or so) and see if the cubes get smaller.
